File: endpoint/templates/go/endpoint.route.go
package <%= nameLowerCase %>routes

import (
	"github.com/<%= username %>/<%= appName %>/server/api/<%= nameLowerCase %>/controller"
	"github.com/julienschmidt/httprouter"
)

func Init(router *httprouter.Router) {
	router.GET("/api/<%= nameLowerCase %>", <%= nameLowerCase %>controller.GetAll)
	router.GET("/api/<%= nameLowerCase %>/:id", <%= nameLowerCase %>controller.GetById)
	router.POST("/api/<%= nameLowerCase %>", <%= nameLowerCase %>controller.New)
	router.PUT("/api/<%= nameLowerCase %>/:id", <%= nameLowerCase %>controller.Update)
	router.DELETE("/api/<%= nameLowerCase %>/:id", <%= nameLowerCase %>controller.Remove)
}

File: endpoint/templates/node/typescript/endpoint.route.ts
/// <reference path="../../../typings/tsd.d.ts" />

"use strict";

import {<%= name %>Controller} from '../controller/<%= name %>.controller';

export class <%= name %>Routes
{
  static init(router:Object)
  {
    router
      .route('/api/<%= nameLowerCase %>')
      .get(<%= name %>Controller.getAll)
      .post(<%= name %>Controller.createNew);

    router
      .route('/api/<%= nameLowerCase %>/:id')
      .delete(<%= name %>Controller.removeById);
  }
}
